js/A step by step guide to functions in JavaScript

```diff
@@ -0,0 +1,213 @@
+Functions are one of the basic concepts in programming. The name itself suggests, it performs a function. A function is a block of code that you can use whenever you need and wherever you required to avoid repeated code blocks. Once a function is written, it can be used over and over and over again. They usually take some input, perform some operations on it and produce some output.  
+
+In this article, we will learn about functions In JavaScript, different ways of defining functions, how hoisting and function scope works and a few more concepts regarding functions. Let's begin.
+
+## The function declaration 
+Functions are defined, or declared, with the **function **keyword. The declaration begins with the function keyword, followed by the name of the function then set of parentheses, which can be used for optional parameters. The code of the function is contained in curly braces.
+
+```javascript
+function nameOfFunction(parameters) {
+    // Code to be executed
+}
+```
+Functions allow a programmer to divide a big program into a number of small and manageable functions.
+For example, If you are building a simple calculator then sum() will be a function. It will take two or more integers as input and return the sum as the result. 
+
+```javascript
+//function declaration
+function sum(a,b){
+return a+b;
+}
+```
+The name of the function can be anything, as long as it follows the same rules as declaring variables.
+JavaScript functions are written in camel case too. It's a best practice to actually tell *what the function is doing *by giving the function name a verb as prefix. Also, the name should describe only one concept.
+
+### 📌 Calling Functions
+Defining a function does not execute it. Defining it simply names the function and specifies what to do when the function is called. Calling the function actually performs the specified actions with the indicated parameters. We can call it by writing the name of the function, followed by parenthesis ().
+
+```javascript
+//function call
+sum(5,3);
+//output : 8
+```
+### 📌 Function Parameters 
+Parameters are input that gets passed into functions as names and behave as local variables. A function can have multiple parameters or no parameters at all.
+If you declared a parameter, but did not pass an argument to it, your parameter would be undefined.
+In JavaScript, function parameters default to *undefined*.
```

Comment:
You said this above so this line is not needed. You can also add this with continuation of above line.  
.. would be undefined. It means that function parameters default to *undefined*.